drivers/display/ls0xx.c

```diff
@@ -0,0 +1,441 @@
+ * @param buf  Buffer containing data to write
+ */
+static int ls0xx_update_display(const struct device *dev,
+				uint16_t start_line,
+				uint16_t num_lines,
+				const uint8_t *buf)
+{
+	struct ls0xx_data *driver = dev->data;
+
+	int err;
+	uint8_t write_cmd;
+	uint16_t max_line;
+
+	LOG_DBG("Lines %d to %d", start_line, start_line + num_lines - 1);
+
+	write_cmd = LS0XX_BIT_WRITECMD | LS0XX_EXTCOMIN_BIT;
+#if defined(CONFIG_LS0XX_VCOM_DRIVER) && !DT_INST_NODE_HAS_PROP(0, extcomin_gpios)
+	extcomin_bit = LS0XX_EXTCOMIN_BIT ? 0x00 : LS0XX_BIT_VCOM;
+#endif
+	tx_buf.len = sizeof(uint8_t);
+	tx_buf.buf =  &write_cmd;
+	err = spi_write(driver->spi_dev, &driver->spi_config, &buf_set);
+	max_line = start_line + num_lines - 1;
+
+	if (buf != NULL) {
+		/* Send each line to the screen including
+		 * the line number and dummy bits
+		 */
+		tx_buf.len = sizeof(uint8_t) * LS0XX_BYTES_PER_LINE;
+		tx_buf.buf = &line_buf[0];
+		for (int l = start_line; l <= max_line; l++) {
+			line_buf[0] = l;
+			memcpy(&line_buf[1], buf, LS0XX_PANEL_WIDTH / LS0XX_PIXELS_PER_BYTE);
+			buf += LS0XX_PANEL_WIDTH / LS0XX_PIXELS_PER_BYTE;
+			err |= spi_write(driver->spi_dev, &driver->spi_config, &buf_set);
+		}
+
+		/* Send another trailing 8 bits for the last line
+		 * These can be any bits, it does not matter
+		 * just reusing the write_cmd buffer
+		 */
+		tx_buf.len = sizeof(uint8_t);
+		tx_buf.buf =  &write_cmd;
+		err |= spi_write(driver->spi_dev, &driver->spi_config, &buf_set);
+	}
+	spi_release(driver->spi_dev, &driver->spi_config);
+
+	return err;
+}
+
+/**
+ * @brief Write to the display at given co-ordinates
+ *
+ * @details Write buffer to the display.
+ *          Buffer width should be a multiple of 8
+ *
+ *          If CONFIG_LS0XX_VCOM_DRIVER=y and extcomin-gpios is not defined
+ *          and buffer is set to NULL, or height/width is 0 then
+ *          only VCOM will be toggled
+ *
+ * @param dev  Pointer to the display device
+ * @param x    X co-ordinate
+ * @param y    Y co-ordinate
+ * @param desc display_buffer_descriptor struct
+ * @param buf  Buffer containing data to write
+ */
+static int ls0xx_write(const struct device *dev, const uint16_t x,
+		       const uint16_t y,
+		       const struct display_buffer_descriptor *desc,
+		       const void *buf)
+{
+
+	LOG_DBG("X: %d, Y: %d, W: %d, H: %d", x, y, desc->width, desc->height);
+
+	if (buf == NULL) {
+#if defined(CONFIG_LS0XX_VCOM_DRIVER) && DT_INST_NODE_HAS_PROP(0, extcomin_gpios)
+		LOG_WRN("Display buffer is not available");
+		return -EINVAL;
+#endif
+	}
+
+	if (desc->width != LS0XX_PANEL_WIDTH) {
+		LOG_ERR("Width not a multiple of %d", LS0XX_PANEL_WIDTH);
+		return -EINVAL;
+	}
+
+	if (desc->pitch != desc->width) {
+		LOG_ERR("Unsupported mode");
+		return -ENOTSUP;
+	}
+
+	if ((y + desc->height) > LS0XX_PANEL_HEIGHT) {
+		LOG_ERR("Buffer out of bounds (height)");
+		return -EINVAL;
+	}
+
+	if ((x + desc->width) > LS0XX_PANEL_WIDTH) {
```

Comment:
Is this actually not just checking if x is zero as line 240 enforces that width equals `LS0XX_PANEL_WIDTH `?

If this is the intent better just check if x == 0.